### PR TITLE
Add cmake support for building against Turing GPU with compute capability 7.5

### PR DESCRIPTION
**Pull request by Char-Aznable**

### CODE REVIEW

File: cmake/kokkos_options.cmake [hunk 1]
```diff
@@ -104,6 +104,7 @@ list(APPEND KOKKOS_ARCH_LIST
      Pascal61        # (GPU) NVIDIA Pascal generation CC 6.1
      Volta70         # (GPU) NVIDIA Volta generation CC 7.0
      Volta72         # (GPU) NVIDIA Volta generation CC 7.2
+     Turing75         # (GPU) NVIDIA Turing generation CC 7.2
```
Comment (dalg24) [inline cmake/kokkos_options.cmake]:
```suggestion
     Turing75         # (GPU) NVIDIA Turing generation CC 7.5
```

Comment (Char-Aznable) [reply cmake/kokkos_options.cmake]: Thanks for the suggestion. Update is force-pushed

**Comment by dalg24** [comment]
> @dalg24 Can you merge this if there's no further issue? Thanks!

I do not have the authority to approve nor merge PRs

**Comment by mhoemmen** [comment]
@crtrott @ibaned @ndellingwood @dsunder @dhollman

**Comment by ibaned** [comment]
```
Running on machine: sems
Repository Status:  724e451838f0a6784c8355721b15dde20f3738c9 Add cmake support for building against Turing GPU with compute capability 7.5


Going to test compilers:  gcc/5.3.0 gcc/7.3.0 intel/17.0.1 clang/4.0.1 cuda/8.0.44
Testing compiler gcc/5.3.0
  Starting job gcc-5.3.0-OpenMP-release
  PASSED gcc-5.3.0-OpenMP-release
Testing compiler gcc/7.3.0
  Starting job gcc-5.3.0-OpenMP-hwloc-release
  PASSED gcc-5.3.0-OpenMP-hwloc-release
  Starting job gcc-7.3.0-Serial-release
  PASSED gcc-7.3.0-Serial-release
Testing compiler intel/17.0.1
  Starting job gcc-7.3.0-Serial-hwloc-release
  PASSED gcc-7.3.0-Serial-hwloc-release
  Starting job intel-17.0.1-OpenMP-release
  PASSED intel-17.0.1-OpenMP-release
Testing compiler clang/4.0.1
  Starting job intel-17.0.1-OpenMP-hwloc-release
  PASSED intel-17.0.1-OpenMP-hwloc-release
  Starting job clang-4.0.1-Pthread_Serial-release
  PASSED clang-4.0.1-Pthread_Serial-release
Testing compiler cuda/8.0.44
  Starting job clang-4.0.1-Pthread_Serial-hwloc-release
  PASSED clang-4.0.1-Pthread_Serial-hwloc-release
  Starting job cuda-8.0.44-Cuda_OpenMP-release
  PASSED cuda-8.0.44-Cuda_OpenMP-release
#######################################################
PASSED TESTS
#######################################################
clang-4.0.1-Pthread_Serial-hwloc-release build_time=190 run_time=270
clang-4.0.1-Pthread_Serial-release build_time=187 run_time=501
cuda-8.0.44-Cuda_OpenMP-release build_time=564 run_time=509
gcc-5.3.0-OpenMP-hwloc-release build_time=181 run_time=121
gcc-5.3.0-OpenMP-release build_time=183 run_time=122
gcc-7.3.0-Serial-hwloc-release build_time=151 run_time=231
gcc-7.3.0-Serial-release build_time=150 run_time=228
intel-17.0.1-OpenMP-hwloc-release build_time=445 run_time=186
intel-17.0.1-OpenMP-release build_time=446 run_time=181
```

**Pull request merged**